### PR TITLE
Fix proposervm.GetAncestors test flake

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

- Fixes test flake seen in: https://github.com/ava-labs/avalanchego/actions/runs/5139214621/jobs/9249412411
- Clarifies usages of `time.Duration`
- Fixes usage of `expected` and `actual` in `require.Equal`
- Removes unnecessary/duplicate error messages

## How this works

The actual fix for the flake was replacing the usage of `ids.Empty.Prefix(...)` with `ids.GenerateTestID()`. This is needed because `ids.GenerateTestID()` internally uses `ids.Empty.Prefix(...)` and could previously result in using the same ID for different blocks (causing a block circle which caused the tests to fail).

## How this was tested

CI

### CODE REVIEW

File: vms/proposervm/batched_vm_test.go [hunk 1]
```diff
@@ -32,9 +32,9 @@ func TestCoreVMNotRemote(t *testing.T) {
 	_, _, proVM, _, _ := initTestProposerVM(t, time.Time{}, 0) // enable ProBlks
 
 	blkID := ids.Empty
-	maxBlocksNum := 1000                            // an high value to get all built blocks
-	maxBlocksSize := 1000000                        // an high value to get all built blocks
-	maxBlocksRetrivalTime := time.Duration(1000000) // an high value to get all built blocks
```
Comment (StephenButtolph) [inline vms/proposervm/batched_vm_test.go:37]: 1 millisecond isn't really that high

File: vms/proposervm/batched_vm_test.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func TestGetAncestorsPreForkOnly(t *testing.T) {
 	// Build some prefork blocks....
 	coreBlk1 := &snowman.TestBlock{
 		TestDecidable: choices.TestDecidable{
-			IDV:     ids.Empty.Prefix(111),
+			IDV:     ids.GenerateTestID(),
```
Comment (StephenButtolph) [inline vms/proposervm/batched_vm_test.go:59]: These changes are the real fix

File: vms/proposervm/batched_vm_test.go [hunk 3]
```diff
@@ -160,11 +160,11 @@ func TestGetAncestorsPreForkOnly(t *testing.T) {
...
 
 	// ... and check returned values are as expected
-	require.NoError(err, "Error calling GetAncestors: %v", err)
-	require.Len(res, 3, "GetAncestor returned %v entries instead of %v", len(res), 3)
-	require.Equal(res[0], builtBlk3.Bytes())
-	require.Equal(res[1], builtBlk2.Bytes())
-	require.Equal(res[2], builtBlk1.Bytes())
+	require.NoError(err)
+	require.Len(res, 3)
+	require.Equal(builtBlk3.Bytes(), res[0])
```
Comment (StephenButtolph) [inline vms/proposervm/batched_vm_test.go:165]: We should always be passing (expected, actual)

File: vms/proposervm/batched_vm_test.go [hunk 4]
```diff
@@ -160,11 +160,11 @@ func TestGetAncestorsPreForkOnly(t *testing.T) {
 	)
 
 	// ... and check returned values are as expected
-	require.NoError(err, "Error calling GetAncestors: %v", err)
```
Comment (StephenButtolph) [inline vms/proposervm/batched_vm_test.go:163]: The error is already logged by the library

**Review by darioush — APPROVED** [review]

**Review by joshua-kim — APPROVED** [review]

**Pull request merged**